doc/man7/EVP_PKEY-DSA.pod

```diff
@@ -10,9 +10,60 @@ EVP_PKEY-DSA, EVP_KEYMGMT-DSA, EVP_PKEY-DH, EVP_KEYMGMT-DH
 The B<DSA> and B<DH> keytypes are implemented in OpenSSL's default and FIPS
 providers.
 The implementations support the basic DSA and DH keys, containing the public
-and private keys I<pub> and I<priv> as well as the three domain parameters
+and private keys I<pub> and I<priv> as well as the three main domain parameters
 I<p>, I<q> and I<g>.
 
+Finite field cryptography (FFC) is a method of implementing discrete logarithm
+cryptography using finite field mathematics. DSA is an example of FFC and
+Diffie-Hellman key establishment algorithms specified in SP800-56A can also be
+implemented as FFC.
+
+For B<DH> FFC key agreement two classes of domain parameters can be used:
+“safe” domain parameters that are associated with approved named safe-prime
```

Comment:
Suggestion: ditch _approved_.

The rationale is that they are only approved by NIST rather than globally.

Reply:
Consider how you recently added "approved" in the ECX key generation PR, I don't quite understand why we should ditch that here...

Reply:
It's only a suggestion.  Feel free to ignore it :)